xdsl/backend/csl/print_csl.py

```diff
@@ -328,6 +330,25 @@ def print_block(self, body: Block):
                     self.print("return;")
                 case csl.ReturnOp(ret_val=val) if val is not None:
                     self.print(f"return {self._get_variable_name_for(val)};")
+                case scf.If(
+                    cond=cond,
+                    output=outputs,
+                    true_region=true_region,
+                    false_region=false_region,
+                ):
+                    for o in outputs:
+                        self.print(f"{self._var_use(o, 'var')};")
+                    with self.descend(
+                        f"if ({self._get_variable_name_for(cond)})"
+                    ) as inner:
+                        inner.print_block(true_region.block)
+                    if false_region:
+                        if len(outputs) > 0 or not (
+                            len(false_region.block.ops) == 1
+                            and isinstance(false_region.block.first_op, scf.Yield)
```

Comment:
Do we need this conditional? worst case scenario it just prints an empty `else {}` block.

Reply:
@dk949 does it slightly differently in `scf.For` loops - basically, instead of passing a value to the `scf.Yield` op, it is assigned to the outside var directly. I'm matching this for now. Happy to hear your thoughts on this?

Reply:
We must avoid at all costs printing an empty else. Yikes. 😉